test/HtmlParser.unittest.js

```
@@ -442,4 +442,84 @@ describe("HtmlParser", () => {
 			expect(requests).not.toContain("./skip");
 		});
 	});
+
+	describe("script type routing", () => {
+		// Every executable spelling must reach the JS pipeline: one that is read as
+		// a data block instead is served to the browser unbundled, which then
+		// executes it. Driven per entry so a dropped or mistyped one fails here.
+		const { JAVASCRIPT_SCRIPT_TYPES } = require("../lib/html/data");
```

ESLint

[error] Missing file extension for "../lib/html/data"

(import/extensions)